lib/wanda/catalog.ex

```diff
@@ -22,54 +24,70 @@ defmodule Wanda.Catalog do
     |> Path.join("/*")
     |> Path.wildcard()
     |> Enum.map(&Path.basename(&1, ".yaml"))
-    |> Enum.map(&get_check(&1))
+    |> get_checks()
   end
 
   @doc """
   Get a check from the catalog.
   """
-  @spec get_check(String.t()) :: Check.t()
+  @spec get_check(String.t()) :: {:ok, Check.t()} | {:error, any}
   def get_check(check_id) do
-    get_catalog_path()
-    |> Path.join("#{check_id}.yaml")
-    |> YamlElixir.read_from_file!()
-    |> map_check()
+    case get_catalog_path()
+         |> Path.join("#{check_id}.yaml")
+         |> YamlElixir.read_from_file() do
+      {:ok, file_content} ->
+        map_check(file_content)
+
+      {:error, reason} = error ->
+        Logger.error("Error getting Check with ID #{check_id}: #{inspect(reason)}")
+        error
+    end
   end
 
   @doc """
   Get specific checks from the catalog.
   """
   @spec get_checks([String.t()]) :: [Check.t()]
   def get_checks(checks_id) do
-    Enum.map(checks_id, &get_check/1)
+    Enum.flat_map(checks_id, fn check_id ->
+      case get_check(check_id) do
+        {:ok, check} -> [check]
+        {:error, _} -> []
+      end
+    end)
   end
 
   defp get_catalog_path do
     Application.fetch_env!(:wanda, Wanda.Catalog)[:catalog_path]
   end
 
-  defp map_check(
-         %{
-           "id" => id,
-           "name" => name,
-           "group" => group,
-           "description" => description,
-           "remediation" => remediation,
-           "facts" => facts,
-           "expectations" => expectations
-         } = check
-       ) do
-    %Check{
-      id: id,
-      name: name,
-      group: group,
-      description: description,
-      remediation: remediation,
-      severity: map_severity(check),
-      facts: Enum.map(facts, &map_fact/1),
-      values: map_values(check),
-      expectations: Enum.map(expectations, &map_expectation/1)
-    }
+  defp map_check(check) do
```

Comment:
Let's use the power of pattern matching here!
So, restore back the `map_check` to the way it was, and simply create a new line after the original (this is important, to keep proper precendence of functions):
```
def map_check(_)  do
  Logger.error("Check with ID #{check_id} is malformed. Check if all the needed fields are present.")
  {:error, :malformed_check}
end
```

This way, if the 1st function doesn't have the expected arguments, it will go to the next.